tests/mtouch/MTouch.cs

```diff
@@ -2251,6 +2251,28 @@ public void MT2015 ()
 		}
 
 		[Test]
+		public void MT2017 ()
+		{
+			using (var mtouch = new MTouchTool ()) {
+				// Create a library, copy it to a different directory, and then
+				// pass both as -r:.. to mtouch. Due to assembly resolution being cached,
+				// this will *not* show the MT2017 error (in fact I don't know if it's possible
+				// to run into MT2017 at all).
```

Comment:
It should not be possible (2017) as the resolver only returns a single (or no) assembly.

In fact there should not be a _real_ reload since the resolver has a cache, so the `Load` should be using the already present `AssemblyDefinition`.

What was failing (without) and that this PR can fix ?